tools/build-tools/src/test/typeValidator/classValidator.spec.ts

```diff
@@ -49,92 +49,503 @@ describe("Class", () => {
+                    this.param2 = param2;
+                }
+            }
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.none);
+        });
+    });
+
+    describe("properties", () => {
+        // changing private properties has no effect on the class API
+        it("changed private properties", () => {
+            const classOld =
+            `
+            export class TestClass {
+                public prop1 = 1;
+                private prop2 = 2;
+            }
+            `;
+
+            const classNew =
+            `
+            export class TestClass {
+                public prop1 = 1;
+                private prop3 = "alpha snorlax";
+            }
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.none);
+        });
+    });
+
+    describe("heritage", () => {
+        // extends changes are breaking
+        it("changed class extensions", () => {
+            const classOld =
+            `
+            class BaseClass<T> extends Promise<T> {}
+            export class TestClass extends BaseClass<string> {}
+            `;
+
+            const classNew =
+            `
+            class BaseClass<T> extends Promise<T> {}
+            export class TestClass extends BaseClass<number> {}
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // structurally identical extends changes with different names are breaking
+        // (this isn't actually breaking but is difficult to handle and errs on the side
+        // of over-bumping)
+        it("changed class extensions that are structurally identical", () => {
+            const classOld =
+            `
+            class BaseClass<T> extends Promise<T> {}
+            export class TestClass extends BaseClass<string> {}
+            `;
+
+            const classNew =
+            `
+            class BaseClass2<T> extends Promise<T> {}
+            export class TestClass extends BaseClass2<string> {}
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // changing implementations is breaking regardless of structural identicality
+        // (this isn't actually breaking but is difficult to handle and errs on the side
+        // of over-bumping)
+        it("changed class implementations that are structurally identical", () => {
+            const classOld =
+            `
+            interface TestInterface {}
+            export class TestClass implements TestInterface {}
+            `;
+
+            const classNew =
+            `
+            interface TestInterface2 {}
+            export class TestClass implements TestInterface2 {}
```

Comment:
Would the results be different if the original interface was included?

```
            interface TestInterface {}
            interface TestInterface2 {}
            export class TestClass implements TestInterface2 {}
```

Reply:
Or

```ts
            interface TestInterface {}
            interface TestInterface2 extends TestInterface {}
            export class TestClass implements TestInterface2 {}
```

Reply:
under the current way of checking things this should still be considered breaking.
`TestInterface2` has a breaking change because its heritage changed (regardless of if it itself changed structurally, due to the note about the overbumping).  Then transitively `TestClass` gets at least the same level of break because `TestInterface2` exists in its heritage